Userland/Libraries/LibWeb/DOM/HTMLCollection.cpp

```diff
@@ -48,6 +48,30 @@ void HTMLCollection::visit_edges(Cell::Visitor& visitor)
     Base::visit_edges(visitor);
     visitor.visit(m_root);
     visitor.visit(m_cached_elements);
+    if (m_cached_name_to_element_mappings)
+        visitor.visit(*m_cached_name_to_element_mappings);
+}
+
+void HTMLCollection::update_name_to_element_mappings_if_needed() const
+{
+    update_cache_if_needed();
+    if (m_cached_name_to_element_mappings)
```

Comment:
Should this be checking if it is non-empty? :thonk:

`update_cache_if_needed` only seems to clear the list, it doesn't null it

Reply:
`m_cached_name_to_element_mappings.clear();` is `OwnPtr::clear()`, but I agree that's ambiguous. Let me replace it with `= nullptr` :)

Reply:
Oh, right! `m_cached_name_to_element_mappings.clear();` vs ` m_cached_name_to_element_mappings->clear();`. 😅